### DISCUSSION TITLE
Moved to M1 - can/should I reinstall?

### DISCUSSION THREAD

**Discussion by arafel** [Homebrew/discussions#1243] [Everyday usage] [answered]
Hi all

I've just moved from an Intel-based MacBook Pro to an M1 based Air. I used the migration assistant, which copied everything over, so I've still got all the brew-based stuff as x86_64. I'd like to reduce dependence on Rosetta (for performance as much as anything). 

The 3.0.0 announcement says "Apple Silicon is now officially supported for installations in /opt/homebrew" - is the best way to migrate simply to uninstall the /usr/local/ version and re-install brew?

Will Homebrew opportunistically rebuild supported packages for M1 as they're updated? Or does it stick with the architecture they were initially installed on?

Thanks
Paul

**Comment by SMillerDev** [comment]
> Or does it stick with the architecture they were initially installed on?

It will because a native brew install is completely separate from an arm one. `/usr/local` will only ever have x86 based software, if you install a homebrew instance in a native terminal it'll install to `/opt/Homebrew` and everything there will be arm native.

**Reply by arafel** [reply]
Okay, that makes sense - thanks.

**Comment by carlocab** [comment, marked as answer]
The short answer is that you can keep the Intel installation, but you may wish to uninstall the Intel versions of formulae that are already bottled on ARM and reinstall a native version.

You can have a look at these threads for more information.

https://github.com/Homebrew/discussions/discussions/417
https://github.com/Homebrew/discussions/discussions/437
https://github.com/Homebrew/discussions/discussions/844
https://github.com/Homebrew/discussions/discussions/531
https://github.com/Homebrew/discussions/discussions/149
https://github.com/Homebrew/discussions/discussions/971
https://github.com/Homebrew/discussions/discussions/727
https://github.com/Homebrew/discussions/discussions/545

**Reply by arafel** [reply]
Thanks Carlo. Clearly my search skills are letting me down, I didn't see those issues. I think I was probably using the wrong keywords. I'll go read those now.

Thanks both for the *really* quick responses!
